src/eos-dropbox-app

```diff
@@ -182,56 +226,32 @@ class DropboxLauncher():
+                                             'org.freedesktop.portal.OpenURI',
+                                             None)
         try:
-            bus = Gio.bus_get_sync(Gio.BusType.SESSION, None)
-            proxy = Gio.DBusProxy.new_sync(bus, Gio.DBusProxyFlags.NONE,
-                                           None,
-                                           'org.freedesktop.portal.Desktop',
-                                           '/org/freedesktop/portal/desktop',
-                                           'org.freedesktop.portal.OpenURI',
-                                           None)
-
             logging.info("Opening Dropbox directory at {}...".format(path))
-            handle = proxy.call_with_unix_fd_list('OpenFile',
-                                                  GLib.Variant('(sha{sv})',
-                                                               ('', 0, GLib.Variant('a{sv}', None))),
-                                                  Gio.DBusCallFlags.NONE,
-                                                  -1,
-                                                  Gio.UnixFDList.new_from_array([fd]))
-
-            # The OpenURI method returns a handle to a 'request object', which stays
-            # alive for the duration of the user interaction related to the method call,
-            # so we connect to its Response signal to know when it's all over.
-            bus.signal_subscribe('org.freedesktop.portal.Desktop',
-                                 'org.freedesktop.portal.Request',
-                                 'Response',
-                                 handle,
-                                 None,
-                                 Gio.DBusSignalFlags.NO_MATCH_RULE,
-                                 self._responseReceived)
-
-        except GLib.Error as e:
-            self._exitOnError("Could not launch the OpenURI portal for {}: {}".format(uri, e.message))
-
-    def _responseReceived(self, connection, sender, path, interface, signal, params):
-        # Format string for the Response signal is 'ua{sv}', but we are
-        # only interested in the first parameter here (response code).
-        #
-        # Response code values:
-        #  0: Success
-        #  1: User cancelled action (e.g. Canceled the App Chooser dialog).
-        #  2: Error
-        (response_code, results) = params.unpack()
-        if response_code == 0:
-            logging.info("OpenURI portal: success!")
-        elif response_code == 1:
-            logging.info("OpenURI portal: cancelled by the user")
-        elif response_code == 2:
-            logging.warning("OpenURI portal: An error happened")
-
+            self._proxy.call_with_unix_fd_list_sync('OpenFile',
+                                                    GLib.Variant('(sha{sv})',
+                                                                 ('', 0, GLib.Variant('a{sv}', None))),
```

Comment:
Same here, you can spell the empty dictionary as `{}` or `None`

Reply:
Ok